packages/core/src/physics/Collider.ts

```diff
@@ -0,0 +1,100 @@
+import { Component } from "../Component";
+import { ignoreClone } from "../clone/CloneManager";
+import { ICollider } from "@oasis-engine/design";
+import { Quaternion, Vector3 } from "@oasis-engine/math";
+import { ColliderShape } from "./shape/ColliderShape";
+import { DisorderedArray } from "../DisorderedArray";
+import { UpdateFlag } from "../UpdateFlag";
+import { Entity } from "../Entity";
+
+/**
+ * Abstract class for collider shapes.
+ */
+export abstract class Collider extends Component {
+  /** @internal */
+  @ignoreClone
+  _index: number = -1;
+  /** @internal */
+  _nativeStaticCollider: ICollider;
+
+  private _shapes: DisorderedArray<ColliderShape> = new DisorderedArray();
+  private _updateFlag: UpdateFlag;
+
+  protected _position: Vector3 = this.entity.transform.position;
+  protected _rotation: Quaternion = this.entity.transform.rotationQuaternion;
```

Comment:
Maybe should use 'this.entity.transform.worldPosition;' and 'this.entity.transform.worldRotationQuaternion'.

Reply:
done